docs/basics/primitives.md

```diff
@@ -5,30 +5,195 @@ to describe the set of types that are native to its platform, we leverage and ov
 
+Several concepts have a very direct translation into Cassandra:
 
 #### Tuples
 
 A `scala.Tuple` can be directly mapped to a `Cassandra.Tuple`. The tuple concept excepts natively within Cassandra and
  we can therefore produce directly translatable mappings between a Scala type and a Cassandra type.
- 
- 
+
+Examples of such translations are visible below:
+
+ | phantom columns                      | Java/Scala type                             | Cassandra type    |
+ | ---------------                      |-------------------                          | ----------------- |
+ | Col[(String, String)]                | Tuple2[String, String]                      | tuple<text, text> |
+ | Col[(String, Int)]                   | Tuple2[String, Int]                         | tuple<text, int> |
+ | Col[(String, Int, List[BigDecimal])] | Tuple3[String, Int, List[BigDecimal]]       | tuple<text, int, frozen<list<decimal>>> |
+
+As seen in the third example, freezing of types should happen automatically during schema auto-generation, and
+this is valid for any kind of type that requires freezing, including:
+
+- Nested collection types.
+- Primary collection or tuple types.
+- UDT types in phantom pro.
+- Auto-derived primitives in phantom-pro.
+
+This also makes it possibly to derive arbitrary encodings for very complex types automatically.
+This feature is only available in phantom-pro using the `autotables` module.
+
+```scala
+import com.outworkers.phantom.auto._
+
+case class SubRecord(
+  id: UUID,
+  description: String,
+  tags: Set[String],
+  timestamp: DateTime
+)
+
+object SubRecord {
+  implicit val subRecordPrimitive: Primitive[SubRecord] = Primitive.auto[SubRecord].derive
+}
+```
+
+This would allow you to use `SubRecord` transparently as a natively Cassandra type
+and all necessary marshalling is generated at compile time with 0 runtime overhead using
+advanced macro machinery and you could arbitrarily use `SubRecord` as a valid column type.
+
+This is an extremely fast operation that will allow you to magically encode complex
+Scala types in Cassandra while maintaining high throughput.
+
+```scala
+
+case class Record(
+  id: UUID,
+  name: String,
+  sub: SubRecord,
+  timestamp: DateTime
+)
+
+abstract class MyTable extends Table[MyTable, Record] {
+  object id extends UUIDColumn
+  object name extends StringColumn
+  object sub extends Col[SubRecord]
+  object timestamp extends DateTimeColumn
+}
+```
+
+
 #### Deriving new primitives from existing ones
 
-Ok, so phantom comes with a pre-defined set that helps with the most common scenarios that are well known and
+Phantom comes with a pre-defined set that helps with the most common scenarios that are well known and
 loved in Cassandra. But what if you want to roll your own type? Well, you can obviously choose to fully implement
 the primitive trait though in some scenarios it's often just easier to simply leverage an existing implementation.
 
```

Comment:
remove paragraph:
For example, if you have the type `case class Test(value: String)` you can use `Primitive.derive` to produce a new implicit primitive in scope
that helps educate Phantom and simultaneously Cassandra of what your type `means`.